### PR TITLE
Delete configuration database on version change

### PR DESCRIPTION
**Pull request by mellon85**
### What does this PR do?

In case there's a datadog-agent version change we have to reset the remote configuration database to ensure it's consistent with our embedded root certificates.

To test it either create a database without the metadata file or put one with a different version than the current one

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/remote/service/util.go [hunk 1]
```diff
@@ -18,19 +18,83 @@ import (
...
-		if db, err = bbolt.Open(path, 0600, &bbolt.Options{}); err != nil {
-			return nil, err
+
+		metadata := new(AgentMetadata)
+		err := json.Unmarshal(metaFile, metadata)
+		if err != nil {
+			return err
+		}
+
+		if metadata.Version == version.AgentVersion {
```
Comment (arbll) [inline pkg/config/remote/service/util.go:82]:
```suggestion
		if metadata.Version != version.AgentVersion {
```

File: pkg/config/remote/service/util.go [hunk 2]
```diff
@@ -18,19 +18,83 @@ import (
...
+			return fmt.Errorf("Database needs to be cleared")
```
Comment (arbll) [inline pkg/config/remote/service/util.go:83]: This is pretty unclean IMO and makes us log at the `error` level while this is intended behavior. We can still log when we clear but at the info level.

File: pkg/config/remote/service/util.go [hunk 3]
```diff
@@ -18,19 +18,83 @@ import (
...
 
+const metaBucket = "meta"
+const metaFile = "meta.json"
+
+type AgentMetadata struct {
+	Version string `json:"version"`
+}
+
+func reopen(path string) (*bbolt.DB, error) {
+	log.Infof("Clear configuration database")
```
Comment (arbll) [inline pkg/config/remote/service/util.go:33]:
```suggestion
	log.Infof("Clear remote configuration database")
```

File: pkg/config/remote/service/util.go (hunk 2 above)
Comment (arbll) [reply pkg/config/remote/service/util.go:83]: (And the info log is already present in reopen)

File: pkg/config/remote/service/util.go [hunk 4]
```diff
@@ -18,19 +18,83 @@ import (
...
+			Version: version.AgentVersion,
+		})
+		if err != nil {
+			return err
+		}
+		return bucket.Put([]byte(metaFile), metaData)
+	})
+}
+
 func openCacheDB(path string) (*bbolt.DB, error) {
```
Comment (arbll) [inline pkg/config/remote/service/util.go:60]: Could you add an unit test checking the this new behavior ?

File: pkg/config/remote/service/util.go (hunk 1 above)
Comment (mellon85) [reply pkg/config/remote/service/util.go]: damn I forgot to commit the fix to that during my test :(

**Review by BaptisteFoy — APPROVED** [review]

**Pull request merged**